client/my-sites/checkout/composite-checkout/composite-checkout.tsx

```diff
@@ -560,15 +563,11 @@ export default function CompositeCheckout( {
 		} )
 	) {
 		debug( 'rendering empty cart page' );
-		const goToPlans = () => {
+		const goToPreviousPage = () => {
 			recordEvent( {
 				type: 'EMPTY_CART_CTA_CLICKED',
 			} );
-			if ( updatedSiteSlug ) {
-				page( `/plans/${ updatedSiteSlug }` );
-			} else {
-				page( '/plans' );
-			}
+			page( previousPath );
```

Comment:
I haven't tested this yet, but I think I already see a possible issue: `getPreviousPath` can return an empty string in some cases (we've encountered a similar issue before with the [checkout close button](https://github.com/Automattic/wp-calypso/blob/f16d1b44bbd32d33d880af689f737ce213c26f99/client/layout/masterbar/checkout.tsx#L32-L76)) so we may need to have a fallback if that happens. Otherwise this redirect probably won't do anything, right? Unless maybe `page('')` brings you to the home page, but if so I think that we still should not rely on an implicit redirect.

I also wonder what this will do for sites where the previous page was a Jetpack site URL? I imagine it won't do anything... but I'm not sure.

Reply:
Great idea, this function has been implemented in composite-checkout https://github.com/Automattic/wp-calypso/blob/48c53458c6cff4404861aab5688ebcae2af38346/client/my-sites/checkout/composite-checkout/composite-checkout.tsx#L560-L639